x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertGenUtilsTests.java

```diff
@@ -100,4 +105,31 @@ public void testIsAnyLocalAddress() throws Exception {
         verifyNoMoreInteractions(address);
     }
 
+    public void testIssuerCertSubjectDN() throws Exception {
+        final ZonedDateTime notBefore = LocalDateTime.now().atZone(ZoneId.of("UTC"));
+        final ZonedDateTime notAfter = LocalDateTime.of(2099, 12, 31, 23, 59, 59, 999999999).atZone(ZoneId.of("UTC"));
+
+        // root CA
+        final X500Principal rootCaPrincipal = new X500Principal("DC=example.com");
+        final KeyPair rootCaKeyPair = CertGenUtils.generateKeyPair(2048);
+        final X509Certificate rootCaCert = CertGenUtils.generateSignedCertificate(rootCaPrincipal, null, rootCaKeyPair,
+            null, rootCaKeyPair.getPrivate(), true, notBefore, notAfter, null);
+
+        // sub CA
+        final X500Principal subCaPrincipal = new X500Principal("DC=Sub CA,DC=example.com");
+        final KeyPair subCaKeyPair = CertGenUtils.generateKeyPair(2048);
+        final X509Certificate subCaCert = CertGenUtils.generateSignedCertificate(subCaPrincipal, null, subCaKeyPair,
+            rootCaCert, rootCaKeyPair.getPrivate(), true, notBefore, notAfter, null);
+
+        // end entity
+        final X500Principal endEntityPrincipal = new X500Principal("CN=TLS Client\\+Server,DC=Sub CA,DC=example.com");
+        final KeyPair endEntityKeyPair = CertGenUtils.generateKeyPair(2048);
+        final X509Certificate endEntityCert = CertGenUtils.generateSignedCertificate(endEntityPrincipal, null, endEntityKeyPair,
+            subCaCert, subCaKeyPair.getPrivate(), true, notBefore, notAfter, null);
+
+        // verify generateSignedCertificate performed DN chaining correctly
+        assertEquals(subCaCert.getSubjectX500Principal(), endEntityCert.getIssuerX500Principal());
+        assertEquals(rootCaCert.getSubjectX500Principal(), subCaCert.getIssuerX500Principal());
+        assertEquals(rootCaCert.getSubjectX500Principal(), rootCaCert.getIssuerX500Principal());
```

Comment:
We generally prefer to use Hamcrest style assertions (although we also have a lot of older code that uses the standard JUnit assertion style).
Matcher-based assertions give much better failure messages which has an enormous benefit over a large codebase like ES (the person triaging a failure has typically not seen the test before, and the better the assertion failure message is, the easier & quicker it is to resolve).

 ```suggestion
        assertThat(rootCaCert.getIssuerX500Principal(), Matchers.equalTo(rootCaCert.getSubjectX500Principal()));
```
(though you would also want to static import `Matchers.equalTo`, but :octocat: suggestions don't make that easy to show).

Reply:
I switched to Matcher.equalsTo().

The updates pass the style check.
```
./gradlew :x-pack:plugin:security:cli:checkstyleTest
```